### DISCUSSION TITLE
Change repository path

### DISCUSSION THREAD

**Discussion by miseeo** [community/community#40812] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

> <img width="420" alt="image" src="https://user-images.githubusercontent.com/108726251/205603908-050f4ccd-7a27-4e9f-8072-a826add93f7c.png">

I connected the local repository to Desktop/ and pushed the bottom second folder, forgetting it. 
The repository appears as Desktop/gitsign/signup/, so can I change the path while keeping commit alive? I've already committed several times and I don't want to make it again from the beginning, so I'm asking you a question.
I'm going to merge with another branch, wouldn't there be a problem if I merge with this route? I'm going to move to one folder in another branch.

**Comment by daniel-cues** [comment]
### Short answer: 
YES.
### Long answer: 
Yes, Git doesn't really care what the directory structure is on the local machine, as long as you keep the .git files in it. You can move around files and directories freely. However, if you'd also like to start a remote repo from scratch keeping the history, you can follow [this](https://www.atlassian.com/git/tutorials/git-move-repository)

Edit: Re-reading your post I believe you want your repository to look like this:
```
Desktop/signup
README.md
```

To do so you need to move the file to the parent folder, with `mv ./gitsign/signup .` (use the wildcard * if you have several files)
And then remove the gitsign directory like so: `git rm -r gitsign` (this will also delete everything in the folder so don't skip step # 1)
Add changes with `git add *`
Finally, commit the changes with `git commit`
And push them to GitHub with `git push`

**And that's it! Your repository is organised now.**

**Reply by daniel-cues** [reply]
He @miseeo did you manage to change the path? If so, feel free to tag this as answered!

**Comment by airtower-luna** [comment, marked as answer]
You can move (rename) files or directories with [`git mv`](https://git-scm.com/docs/git-mv). Then commit, push, and you're done. The web UI doesn't have a way to rename directories as far as I know.

> The repository appears as Desktop/gitsign/signup/, so can I change the path while keeping commit alive?

What do you mean by keeping a commit "alive"? The Git history is persistent unless you explicitly change it, so your old commits will always be available.
